### DISCUSSION TITLE
Athena++ globaldisk simulation does not seem to accrete

### DISCUSSION THREAD

**Discussion by blazar1** [PrincetonUniversity/athena#610] [Q&A]
I'm a relatively new user with a potentially basic question. I attempted to run gloabldisk.cpp problem generator without radiation (as a trial and to learn) but after a very long time (order of 100 fiducial times) the initial torus seems to just stay there and did not accrete towards the black hole, as indicated by azimuthally averaged density snap shots. This occurs regardless of initial density, which I tried to use parameters from Huang et al. (2023): https://iopscience.iop.org/article/10.3847/1538-4357/acb6fc

I imagine I'm missing something basic here?

**Comment by tomidakn** [comment]
The problem generator in disk.cpp does not have magnetic fields, so there is no mechanism to drive accretion. Did you write your own problem generator to add magnetic fields, or are you referring something else?

**Reply by blazar1** [reply]
Thank you for replying! I ran globaldisk.cpp (attached), which does have magnetic field with single and multi-loop initial configuration options.
[globaldisk.cpp.txt](https://github.com/user-attachments/files/16499606/globaldisk.cpp.txt)

**Reply by tomidakn** [reply]
This pgen is not what we officially provide, so please ask the author of the code. Maybe your field is just too weak, so make sure the initial field has a reasonable strength and distribution.

But at least I spotted one mistake, l.645, the if condition should be 
`if (NON_BAROTROPIC_EOS)` instead of `if(NR_RADIATION_ENABLED)`,
otherwise, the magnetic energy is not correctly calculated if you naively turn off the radiation.
